main/pipeline/pipeline.cc

```diff
@@ -291,6 +304,18 @@ incrementalResolve(core::GlobalState &gs, vector<ast::ParsedFile> what,
             ENFORCE(result.hasResult());
             what = move(result.result());
 
+            if (opts.runLSP && gs.lspQuery.kind == core::lsp::Query::Kind::NONE) {
```

Comment:
I don't want to do extra error flushes if Sorbet does a typecheck for LSPQuery purposes. For example, Sorbet runs an incremental resolver during document symbol request. It broke some tests with either extra results or unnecessary errors, I can't remember exactly. Basically, I wanted Sorbet to extra flush errors if and only if Sorbet works in the LSP mode and it's used for publishing diagnostics.

Reply:
I think this is likely because of the choice to store state on `core::GlobalState::errors` instead of in the `ErrorQueue`/`ErrorFlusher`. When we call `runFastPath`, we re-use the GlobalState object, but we make a **new** error queue and error flusher. That means that when running the fast path, any error messages that get reported accumulate into this new random error queue/flusher pair, and then don't get reported.

But by having the code cache errors on the GlobalState object which lives longer than a single fast path request, running a no-op retypecheck fast path essentially poisons the cache.

Reply:
So I would guess that we would not need these checks with the data modeling changes I mention elsewhere to avoid caching errors on GlobalState.